packages/local/src/main.ts

```
@@ -149,17 +154,14 @@ class LocalTableland {
     // wait until initialization is done
     await waitForReady(registryReadyEvent, this.initEmitter);
 
-    // Deploy the Registry to the Hardhat node
-    logSync(
-      spawnSync(
-        isWindows() ? "npx.cmd" : "npx",
-        ["hardhat", "run", "--network", "localhost", "scripts/deploy.ts"],
-        {
-          cwd: this.registryDir,
-        }
-      ),
-      !inDebugMode()
-    );
+    this._deployRegistry();
+
+    const deployed = await this.#_ensureRegistry();
+    if (!deployed) {
+      throw new Error(
```

This ensures we fail fast and loud if the deploy failed.  At the moment you have to look through the CI logs and see if the failing tests are becaseue the registry wasn't deployed, or for some legitimate reason.